src/typing_extensions.py

```diff
@@ -942,7 +942,13 @@ def __new__(cls, name, bases, ns, *, total=True, closed=False):
                 tp_dict.__orig_bases__ = bases
 
             annotations = {}
-            own_annotations = ns.get('__annotations__', {})
+            if "__annotations__" in ns:
+                own_annotations = ns["__annotations__"]
+            elif "__annotate__" in ns:
+                # TODO: Use inspect.VALUE here, and make the annotations lazily evaluated
+                own_annotations = ns["__annotate__"](1)
+            else:
+                own_annotations = {}
```

Comment:
You mentioned in https://github.com/python/cpython/pull/119361#discussion_r1615277717 that you'd be adding a helper function for this kind of thing at some point -- would it be worth waiting for that, since the first 3.14 alpha is some way off?

I can also see the advantages of getting `typing_extensions` working with your branch, though, so that you can start testing with the rest of the ecosystem that depends on `typing_extensions`. So I definitely don't want to block you merging this.

Reply:
I do expect we'll need more changes later, but it's helpful to get this merged so I can test the rest of the ecosystem too.